rfc/0008-builtin-blocktypes/README.md

```diff
@@ -0,0 +1,212 @@
+<!--
+SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+
+SPDX-License-Identifier: AGPL-3.0-only
+-->
+
+# RFC 0008: Builtin blocktypes
+
+|             |                      |
+|-------------|----------------------|
+| Feature Tag | `builtin-blocktypes` |
+| Status      | `DISCUSSION`              | <!-- Possible values: DRAFT, DISCUSSION, ACCEPTED, REJECTED -->
+| Responsible | `@felix-oq`          |
+<!-- 
+  Status Overview:
+  - DRAFT: The RFC is not ready for a review and currently under change. Feel free to already ask for feedback on the structure and contents at this stage.
+  - DISCUSSION: The RFC is open for discussion. Usually, we open a PR to trigger discussions.
+  - ACCEPTED: The RFC was accepted. Create issues to prepare implementation of the RFC.
+  - REJECTED: The RFC was rejected. If another revision emerges, switch to status DRAFT.
+-->
+
+## Summary
+
+Blocktypes that are built into the language are declared syntactically by denoting IO types and their properties.
+This change does not change the semantics, but it can serve as a foundation for composite block types in an upcoming RFC.
+
+## Motivation
+
+Currently, all blocktypes only exist implicitly in the language.
+To make their definition more explicit, we should support the declaration of builtin blocktypes (i.e. blocktypes that are built into the language).
+This allows users to look up IO types and properties without having to open the documentation pages.
+
+## Explanation
+
+- `builtin` keyword to indicate that the blocktype is built into the language
+  - Produces an error if the blocktype is unknown
+- `input` / `output` keywords to type the input and output of the blocktype
+  - If omitted, a blocktype has no input or output
+- `property` keyword to define a property with a name and a type
+  - Optionally, a default value can be assigned
+- Adds new keywords for existing property types:
+  - Valuetypes: `regex`, `cellrange` / `row` / `column` / `cell`, `valuetype-assignment`
+  - Typed collections: `collection<type>`
+    - Nested collections are not supported for now
+  - IO types: `File`, `FileSystem`, `TextFile`, `Sheet`, `Table`
```

Comment:
Not sure how I feel about having valuetypes and io types having totally distinct writing styles (lower case with dash separator vs uppercase and camelcase). I would merge them to one style we use in the language already for property names.

Reply:
Sure, we can use the same cases :+1:

Then it would be `valuetypeAssignment` instead of `valuetype-assignment`, right?

Reply:
Yes. But also fileSystem for io types.